infra.bs

```diff
@@ -84,6 +85,73 @@ either width or height is zero"), means an inclusive "or" (implying "or both"),
+with which you have an account.
+
+<p>There are other mechanisms that are more subtle. Certain characteristics of a user's system can
+be used to distinguish groups of users from each other; by collecting enough such information, an
+individual user's browser's "digital fingerprint" can be computed, which can be as good, if not
+better, as an IP address in ascertaining which requests are from the same user.
+
+<p>Grouping requests in this manner, especially across multiple sites, can be used for both benign
+(and even arguably positive) purposes, as well as for malevolent purposes. An example of a
+reasonably benign purpose would be determining whether a particular person seems to prefer sites
+with dog illustrations as opposed to sites with cat illustrations (based on how often they visit
+the sites in question) and then automatically using the preferred illustrations on subsequent
+visits to participating sites. Malevolent purposes, however, could include governments combining
+information such as the person's home address (determined from the addresses they use when getting
+driving directions on one site) with their apparent political affiliations (determined by
+examining the forum sites that they participate in) to determine whether the person should be
+prevented from voting in an election.
+
+<p>Since the malevolent purposes can be remarkably evil, user agent implementors are encouraged to
+consider how to provide their users with tools to minimize leaking information that could be used
+to fingerprint a user.
+
+<p>Unfortunately, as the first paragraph in this section implies, sometimes there is great benefit
+to be derived from exposing the very information that can also be used for fingerprinting
+purposes, so it's not as easy as simply blocking all possible leaks. For instance, the ability to
+log into a site to post under a specific identity requires that the user's requests be
+identifiable as all being from the same user, more or less by definition. More subtly, though,
+information such as how wide text is, which is necessary for many effects that involve drawing
+text onto a canvas (e.g. any effect that involves drawing a border around the text) also leaks
+information that can be used to group a user's requests. (In this case, by potentially exposing,
+via a brute force search, which fonts a user has installed, information which can vary
+considerably from user to user.)
+
+<p class=fingerprinting-vector>Features that are defined in documents using the Infra Standard that
+can be <dfn export lt="fingerprinting vector">used to fingerprint the user</dfn> are marked as this
+paragraph is.
+
+<p>Other features in the platform can be used for the same purpose, though, including, though not
+limited to:
+
+<ul>
+ <li>The exact list of which features a user agents supports.
+
+ <li>The maximum allowed stack depth for recursion in script.
+
+ <li>Features that describe the user's environment.
+
+ <li>The user's time zone.
```

Comment:
Time zone is kinda standalone, since I believe it's the main thing that leaks from JavaScript (other than performance characteristics and oh my SharedArrayBuffer). Accept-Language is a good one too.